pkgs/servers/home-assistant/custom-components/tapo-p100/default.nix

```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildHomeAssistantComponent,
+  plugp100,
+}:
+buildHomeAssistantComponent rec {
+  pname = "tapo-p100";
```

Comment:
I didn't know other components can't use same domain
At least in this case I think `tapo` is better.

Reply:
This is because multiple components can't really share the same `custom_components/<domain>/` directory. At least I think that it wouldn't work.

Reply:
The current idea, not yet ratified in any way, is `home-assistant-custom-component."<owner>/<domain>"`.

That would prevent naming collisions in a very easy way, and also expose the `domain` name, of which only one of the same name can be installed at a time.

What does everyone think about this proposal?

Reply:
After giving it some thought, I believe that's a good idea since we shouldn't be limited to adding components only from something like HACS default repository and HACS official repository handles like [this](https://github.com/hacs/default/blob/master/integration) anyways.
I don't know much of the details, just learning now. Looking through [this](https://github.com/home-assistant/developers.home-assistant/issues/570) and [official documentation](https://developers.home-assistant.io/docs/creating_integration_manifest/), it appears that the domain name is used as the component's namespace and must be unique for official components. Though, this may not apply to custom components that can also be installed from custom repositories.

Reply:
I'm working on making this change happen in #269940.